WordPress/Classes/Services/CommentServiceRemoteCoreRESTAPI.swift

```diff
@@ -0,0 +1,323 @@
+            failure?(URLError(.unknown))
+            return
+        }
+
+        Task { @MainActor in
+            do {
+                let comment = try await client.api.comments.update(commentId: commentID.int64Value, params: .init(status: status)).data
+                success?(RemoteComment(comment: comment))
+            } catch {
+                failure?(error)
+            }
+        }
+    }
+
+    func trashComment(_ comment: RemoteComment, success: (() -> Void)?, failure: (((any Error)?) -> Void)?) {
+        guard let commentID = comment.commentID else {
+            wpAssertionFailure("comment id missing in the comment")
+            failure?(URLError(.unknown))
+            return
+        }
+
+        Task { @MainActor in
+            do {
+                // The `WordPressAPI.comments.trash` moves comments to trash.
+                // The `WordPressAPI.comments.delete` deletes comments  permanently.
+                //
+                // Even though this function here is called "trash comment", it's actually used to deleted
+                // comments permanently.
+                _ = try await client.api.comments.delete(commentId: commentID.int64Value, params: .init())
+                success?()
+            } catch {
+                failure?(error)
+            }
+        }
+    }
+}
+
+private extension RemoteComment {
+    convenience init(comment: CommentWithEditContext) {
+        self.init()
+
+        self.commentID = NSNumber(value: comment.id)
+        self.authorID = NSNumber(value: comment.author)
+        self.author = comment.authorName
+        self.authorEmail = comment.authorEmail
+        self.authorUrl = comment.authorUrl
+        self.authorAvatarURL = comment.authorAvatarUrls.avatarURL()?.absoluteString
+        self.authorIP = comment.authorIp
+        self.content = comment.content.raw
+        self.rawContent = comment.content.raw
+        self.date = comment.dateGmt
+        self.link = comment.link
+        self.parentID = NSNumber(value: comment.parent)
+        self.postID = NSNumber(value: comment.post)
+
+        self.status = comment.status.commentStatusType?.description
+        self.type = comment.commentType.type
+
+        // The following properties are not available in .org REST API.
+        self.postTitle = nil
+        self.isLiked = false
+        self.likeCount = 0
+        self.canModerate = false
+    }
+
+    /// When `RemoteComment` is created by the app to send to API, some properties are set to default values (like
+    /// empty string, 0). We should treat those values as `nil`, because wordpress-rs emits nil values in request body,
+    /// but keeps them if they are "" or 0.
+    ///
+    /// We'll treat each property case by case, but in general it's safe to treat "" or 0 as nil. For example, there is
+    /// no post whose id is 0, and there is no email address is an empty string.
+    func resetDefaultValuesToNil() {
```

Comment:
I wonder if we should emit a new type here instead of `RemoteComment`? This feels like a lot of baggage to work around?

I don't feel strongly about this at all, so I'm happy to be told I'm wrong here – just feels like we're swimming against the tide a bit here

Reply:
We have to work with `RemoteComment` because the `CommentServiceRemote` protocol uses it. It does feel like we are swimming against the tide, because we are making the new APIs in `wordpress-rs` to work with the existing types `CommentServiceRemote` & `RemoteComment` in the app. When the day we delete the WordPressKit comes, we might be able to simplify the solutions here, because `wordpress-rs` will become the only API involved.

Reply:
This makes sense – right now we're doing a lot of adaptor work because it makes the interim step easier. Probably a lot easier to delete code later than rework everything now.

Thanks for the explanation!